tests/robustness/watch.go

```diff
@@ -45,35 +46,59 @@ func collectClusterWatchEvents(ctx context.Context, t *testing.T, lg *zap.Logger
 		if err != nil {
 			t.Fatal(err)
 		}
-
+		memberChan := make(chan int64, 1)
+		memberMaxRevisionChans[i] = memberChan
 		wg.Add(1)
 		go func(i int, c *clientv3.Client) {
 			defer wg.Done()
 			defer c.Close()
-			responses := watchMember(ctx, lg, c)
+			responses := watchMember(lg, c, memberChan)
 			mux.Lock()
 			memberResponses[i] = responses
 			mux.Unlock()
 		}(i, c)
 	}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		maxRevision := <-maxRevisionChan
+		for _, memberChan := range memberMaxRevisionChans {
+			memberChan <- maxRevision
+		}
+	}()
 	wg.Wait()
 	return memberResponses
 }
 
-func watchMember(ctx context.Context, lg *zap.Logger, c *clientv3.Client) (resps []watchResponse) {
+func watchMember(lg *zap.Logger, c *clientv3.Client, maxRevisionChan <-chan int64) (resps []watchResponse) {
 	var lastRevision int64 = 0
+	var maxRevision int64 = 0
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	watch := c.Watch(ctx, "", clientv3.WithPrefix(), clientv3.WithRev(lastRevision+1), clientv3.WithProgressNotify())
 	for {
 		select {
 		case <-ctx.Done():
 			return resps
-		default:
-		}
-		for resp := range c.Watch(ctx, "", clientv3.WithPrefix(), clientv3.WithRev(lastRevision+1), clientv3.WithProgressNotify()) {
+		case maxRevision = <-maxRevisionChan:
```

Comment:
After reread it seems we support only 1 value in the maxRevisions. I would assert it explicitly here then.

Reply:
I feel like it is more natural to put `if maxRevision == 0 and then error out` logic in the receiver side.

Reply:
`0` is not correct revision so I never expect it will be written to maxRevisionChan. If it is, function will exit via `context.Done()`. Having one return point via context is clearer.